src/ray/raylet/node_manager.cc

```diff
@@ -1029,7 +1041,7 @@ void NodeManager::HandleDisconnectedActor(const ActorID &actor_id, bool was_loca
     // So if we receive any actor tasks before we receive GCS notification,
     // these tasks can be correctly routed to the `MethodsWaitingForActorCreation`
     // queue, instead of being assigned to the dead actor.
-    HandleActorStateTransition(actor_id, ActorRegistration(new_actor_info));
+    HandleActorStateTransition(actor_id, ActorRegistration(new_actor_info), was_local);
```

Comment:
Can you please remove this flag? Thanks!

Reply:
@stephanie-wang hi，if remove was_local flag, then maybe need in HandleDisconnectedActor call HandleTaskReconstruction other than in HandleActorStateTransition.I am not sure if i have understand you.thanks.

Reply:
Yes, I mean to move the HandleTaskReconstruction call into HandleDisconnectedActor and remove the was_local flag here.

Reply:
hi, @stephanie-wang , I was confused that if call HandleTaskReconstruction  in HandleDisconnectedActor before HandleActorStateTransition, it can not update local registry. If call HandleTaskReconstruction  in HandleDisconnectedActor after HandleActorStateTransition, HandleActorStateTransition has call ListenAndMaybeReconstruct, it is repetitive.

Reply:
It's okay if it doesn't update the local registry immediately. You can just post HandleTaskReconstruction in a separate callback on the event loop from HandleDisconnectedActor.

Mostly, I am trying to avoid any changes to the current actor lifecycle management since all of this will likely be deprecated soon by the direct calls.

Reply:
Thanks, i have fix the code.